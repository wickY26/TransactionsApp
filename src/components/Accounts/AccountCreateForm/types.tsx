import { Account } from '../../../containers/Accounts/types';

export interface AccountCreateFormProps {
  onCreateAccount: (data: Account) => void;
}
