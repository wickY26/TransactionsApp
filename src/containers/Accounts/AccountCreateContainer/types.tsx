import { ReactElement } from 'react';

import { Account } from '../types';

export interface RenderProps {
  onCreateAccount: (data: Account) => void;
}

export interface AccountCreateContainerProps {
  children: (props: RenderProps) => ReactElement;
};
