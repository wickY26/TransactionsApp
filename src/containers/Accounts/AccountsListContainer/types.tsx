import { ReactElement } from 'react';

import { Account } from '../types';

export interface RenderProps {
  accounts: Account[];
}

export interface AccountsListContainerProps {
  children: (props: RenderProps) => ReactElement;
};
