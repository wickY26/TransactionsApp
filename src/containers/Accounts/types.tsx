import { Action } from 'redux';

import { ActionTypes as at } from './constant';

export interface Account {
  name: string;
  surname: string;
  id?: number;
};

export interface FetchAccountsSuccessAction extends Action {
  type: at.FetchAccountsSuccess;
  payload: Account[];
}

export interface FetchAccountsErrorAction extends Action {
  type: at.FetchAccountsError;
}

export interface CreateAccountSuccessAction extends Action {
  type: at.CreateAccountSuccess;
  payload: Account;
}

export interface CreateAccountErrorAction extends Action {
  type: at.CreateAccountError;
}

export interface State {
  accounts: Account[];
}

export type AccountsActions = 
  | FetchAccountsSuccessAction
  | FetchAccountsErrorAction
  | CreateAccountSuccessAction
  | CreateAccountErrorAction
