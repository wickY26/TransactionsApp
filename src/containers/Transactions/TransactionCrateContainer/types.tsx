import { ReactElement } from 'react';

import { TransactionBody } from '../types';

export interface RenderProps {
  onCreateTransaction: (data: TransactionBody) => void;
}

export interface TransactionCreateContainerProps {
  children: (props: RenderProps) => ReactElement;
};
