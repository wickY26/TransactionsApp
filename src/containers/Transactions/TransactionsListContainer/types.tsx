import { ReactElement } from 'react';

import { Transaction } from '../types';

export interface RenderProps {
  transactions: Transaction[];
}

export interface TransactionsListContainerProps {
  children: (props: RenderProps) => ReactElement;
};
