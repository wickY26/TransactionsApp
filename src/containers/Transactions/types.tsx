import { Action } from 'redux';

import { ActionTypes as at } from './constant';

export interface TransactionAccount {
  name: string;
  surname: string;
  id: number;
}

export interface Transaction {
  from: TransactionAccount;
  to: TransactionAccount;
  amount: number;
  description: string;
  id: number;
};

export interface TransactionBody {
  from: number;
  to: number;
  amount: number;
  description: string;
};

export interface FetchTransactionsSuccessAction extends Action {
  type: at.FetchTransactionsSuccess;
  payload: Transaction[];
}

export interface FetchTransactionsErrorAction extends Action {
  type: at.FetchTransactionsError;
}

export interface CreateTransactionSuccessAction extends Action {
  type: at.CreateTransactionSuccess;
  payload: Transaction;
}

export interface CreateTransactionErrorAction extends Action {
  type: at.CreateTransactionError;
}

export interface State {
  transactions: Transaction[];
}

export type TransactionsActions = 
  | FetchTransactionsSuccessAction
  | FetchTransactionsErrorAction
  | CreateTransactionSuccessAction
  | CreateTransactionErrorAction
