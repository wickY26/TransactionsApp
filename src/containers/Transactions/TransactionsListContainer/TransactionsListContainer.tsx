import { FC, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { TransactionsListContainerProps } from './types';
import { thunkFetchTransactions } from '../actions';
import { transactionsSelector } from '../selectors';

const AccountsListContainer: FC<TransactionsListContainerProps> = ({ children }) => {
  const transactions = useSelector(transactionsSelector);
  const dispatch = useDispatch();
 
  useEffect(() => {
    dispatch(thunkFetchTransactions());
  }, [dispatch]);

  return children({ transactions });
};

export default AccountsListContainer;
