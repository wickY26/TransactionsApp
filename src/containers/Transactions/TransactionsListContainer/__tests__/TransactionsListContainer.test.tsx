import React from 'react';
import { render } from '@testing-library/react';

import TransactionsListContainer from '..';
import { Transaction, TransactionAccount } from '../../types';

const mockAccount: TransactionAccount = {name: 'Poyraz', surname: 'Yilmaz', id: 0};
const mockTransaction: Transaction = { from: mockAccount, to: mockAccount, amount: 50, description: 'description', id: 0 };
const mockDispatch = jest.fn();
const mockSelector = jest.fn();
const mockThunkFetchTransaction = jest.fn();

jest.mock('react-redux', () => ({
  useSelector: () => mockSelector(),
  useDispatch: () => mockDispatch,
}));

jest.mock('../../actions', () => ({
  thunkFetchTransactions: () => mockThunkFetchTransaction(),
}));

const renderComponent = () => render(
  <TransactionsListContainer>
    {({transactions}) => (
      <ul>
        {transactions?.map(transaction => <li key={transaction.id}>{transaction.amount}</li>)}
      </ul>
    )}
  </TransactionsListContainer>
);

describe('TransactionsListContainer', () => {

  beforeEach(() => {
    mockSelector.mockReturnValue([mockTransaction]);
    mockThunkFetchTransaction.mockReturnValue('FetchTransactionAction');
  });

  it('should render correctly', () => {
    const component = renderComponent();
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should dispatch fetchTransactionsAction', () => {
    renderComponent();
    expect(mockThunkFetchTransaction).toHaveBeenCalledTimes(1);
    expect(mockDispatch).toHaveBeenCalledWith('FetchTransactionAction');
  });
  
  it('should select transactions from store', () => {
    renderComponent();
    expect(mockSelector).toHaveBeenCalled();
  });

});
