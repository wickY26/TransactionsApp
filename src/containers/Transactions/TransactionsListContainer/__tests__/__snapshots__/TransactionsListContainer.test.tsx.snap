// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionsListContainer should render correctly 1`] = `
<DocumentFragment>
  <ul>
    <li>
      50
    </li>
  </ul>
</DocumentFragment>
`;
